SCClassLibrary/Common/Control/Server.sc

```diff
@@ -511,6 +516,67 @@ Server {
 		^nodeAllocator.freePerm(id)
 	}
 
+	prHandleClientLoginInfoFromServer { |newClientID, newMaxLogins|
+
+		// turn notified off to allow setting clientID
+		statusWatcher.notified = false;
+
+		// only set maxLogins if not internal server
+		if (inProcess.not) {
+			if (newMaxLogins.notNil) {
+				if (newMaxLogins != options.maxLogins) {
+					"%: server process has maxLogins % - adjusting my options accordingly.\n"
+					.postf(this, newMaxLogins);
+				} {
+					"%: server process's maxLogins (%) matches with my options.\n"
+					.postf(this, newMaxLogins);
+				};
+				options.maxLogins = maxNumClients = newMaxLogins;
+			} {
+				"%: no maxLogins info from server process.\n"
+				.postf(this, newMaxLogins);
+			};
+		};
+
+		if (newClientID == clientID) {
+			"%: keeping clientID (%) as confirmed by server process.\n"
+			.postf(this, newClientID);
+		} {
+			"%: setting clientID to %, as obtained from server process.\n"
+			.postf(this, newClientID);
+		};
+		this.clientID = newClientID;
+		statusWatcher.notified = true; // and lock again
+
+		forkIfNeeded {
+			this.changed(\serverRunning);
```

Comment:
@adcxyz : @patrickdupuis pointed out that this line causes an error when booting from the GUI. I was able to fix it by wrapping it in `{ }.defer` but that seems like a hack since no other calls to `changed` are treated that way in this file.

Reply:
the fork should really be ```fork ( { ... }, AppClock)``` - my slop. 
done!